core/chains/evm/gas/fee_history_estimator_test.go

```diff
@@ -0,0 +1,539 @@
+		client := mocks.NewFeeHistoryEstimatorClient(t)
+		client.On("SuggestGasPrice", mock.Anything).Return(big.NewInt(1), nil).Once()
+		originalGasPrice := assets.NewWeiI(10)
+
+		cfg := gas.FeeHistoryEstimatorConfig{BumpPercent: 50}
+
+		maxPrice := assets.NewWeiI(14)
+		u := gas.NewFeeHistoryEstimator(logger.Test(t), client, cfg, chainID, nil)
+		err := u.RefreshGasPrice()
+		assert.NoError(t, err)
+		gas, _, err := u.BumpLegacyGas(tests.Context(t), originalGasPrice, gasLimit, maxPrice, nil)
+		assert.NoError(t, err)
+		assert.Equal(t, maxPrice, gas)
+	})
+
+	t.Run("fails if the bumped gas price is lower than the minimum bump percentage", func(t *testing.T) {
+		client := mocks.NewFeeHistoryEstimatorClient(t)
+		client.On("SuggestGasPrice", mock.Anything).Return(big.NewInt(100), nil).Once()
+		originalGasPrice := assets.NewWeiI(100)
+
+		cfg := gas.FeeHistoryEstimatorConfig{BumpPercent: 20}
+
+		// Price will be capped by the max price
+		maxPrice := assets.NewWeiI(101)
+		u := gas.NewFeeHistoryEstimator(logger.Test(t), client, cfg, chainID, nil)
+		err := u.RefreshGasPrice()
+		assert.NoError(t, err)
+		_, _, err = u.BumpLegacyGas(tests.Context(t), originalGasPrice, gasLimit, maxPrice, nil)
+		assert.Error(t, err)
+	})
+}
+
+func TestFeeHistoryEstimatorGetDynamicFee(t *testing.T) {
+	t.Parallel()
+
+	maxPrice := assets.NewWeiI(100)
+	chainID := big.NewInt(0)
+
+	t.Run("fetches a new dynamic fee when first called", func(t *testing.T) {
+		client := mocks.NewFeeHistoryEstimatorClient(t)
+		baseFee := big.NewInt(5)
+		maxPriorityFeePerGas1 := big.NewInt(33)
+		maxPriorityFeePerGas2 := big.NewInt(20)
+
+		feeHistoryResult := &ethereum.FeeHistory{
+			OldestBlock:  big.NewInt(1),
+			Reward:       [][]*big.Int{{maxPriorityFeePerGas1, big.NewInt(5)}, {maxPriorityFeePerGas2, big.NewInt(5)}}, // first one represents market price and second one connectivity price
+			BaseFee:      []*big.Int{baseFee, baseFee},
+			GasUsedRatio: nil,
+		}
+		client.On("FeeHistory", mock.Anything, mock.Anything, mock.Anything).Return(feeHistoryResult, nil).Once()
+
+		blockHistoryLength := 2
+		cfg := gas.FeeHistoryEstimatorConfig{BlockHistorySize: uint64(blockHistoryLength)}
+		avrgPriorityFee := big.NewInt(0)
+		avrgPriorityFee.Add(maxPriorityFeePerGas1, maxPriorityFeePerGas2).Div(avrgPriorityFee, big.NewInt(int64(blockHistoryLength)))
+		maxFee := (*assets.Wei)(baseFee).AddPercentage(gas.BaseFeeBufferPercentage).Add((*assets.Wei)(avrgPriorityFee))
+
+		u := gas.NewFeeHistoryEstimator(logger.Test(t), client, cfg, chainID, nil)
+		err := u.RefreshDynamicPrice()
+		assert.NoError(t, err)
+		dynamicFee, err := u.GetDynamicFee(tests.Context(t), maxPrice)
+		assert.NoError(t, err)
+		assert.Equal(t, maxFee, dynamicFee.FeeCap)
+		assert.Equal(t, (*assets.Wei)(avrgPriorityFee), dynamicFee.TipCap)
+	})
+
+	t.Run("fails if BlockHistorySize is zero and tries to fetch new prices", func(t *testing.T) {
```

Comment:
Is this test still relevant? I think this is probably still returning an error because the price isn't being refreshed/set before `GetDynamicFee` is called